projects/com.oracle.truffle.llvm.nodes.impl/src/com/oracle/truffle/llvm/nodes/impl/control/LLVMSwitchNode.java

```diff
@@ -130,6 +129,7 @@ public LLVMI8ProfilingSwitchNode(LLVMI8Node cond, LLVMI8Node[] cases, int[] succ
         }
 
         @Override
+        @ExplodeLoop
         boolean profile(int i, boolean value) {
             return profiles[i].profile(value);
         }
```

Comment:
This use of explode loop still doesn't make sense. There's no loop here! Maybe you have a misunderstanding about what the annotation does? For example it's not transient to called methods as well, if that's what you think.